pulsar-broker-common/src/main/java/org/apache/pulsar/broker/authorization/PulsarAuthorizationProvider.java

```diff
@@ -115,7 +115,11 @@ public CompletableFuture<Boolean> canConsumeAsync(TopicName topicName, String ro
                             // list is empty)
                             Set<String> roles = policies.get().auth_policies
                                     .getSubscriptionAuthentication().get(subscription);
-                            if (roles != null && !roles.isEmpty() && !roles.contains(role)) {
+                            Map<String, Set<AuthAction>> namespaceRolesAuth =
+                                    policies.get().auth_policies.getNamespaceAuthentication();
+                            if (!(namespaceRolesAuth != null && namespaceRolesAuth.containsKey(role)
+                                    && namespaceRolesAuth.get(role).contains(AuthAction.consume))
```

Comment:
Should be `!namespaceRolesAuth.get(role).contains(AuthAction.consume)` ?

Reply:
And I also find there is no `return` at line 139. We should add `return CompletableFuture.completedFuture(true)`

Reply:
> Should be `!namespaceRolesAuth.get(role).contains(AuthAction.consume)` ?

we should check the response for function: `namespaceRolesAuth.get(role)`, maybe the response is null.
the check maybe the follow:
```java
                            if (!(namespaceRolesAuth.containsKey(role)
                                    && namespaceRolesAuth.get(role).contains(AuthAction.consume))
                                    && roles != null && !roles.isEmpty() && !roles.contains(role)) {
                                log.warn("[{}] is not authorized to subscribe on {}-{}", role, topicName, subscription);
                                return CompletableFuture.completedFuture(false);
                            }
```


> And I also find there is no `return` at line 139. We should add `return CompletableFuture.completedFuture(true)`

at line 139, we are not need to add return, because after check subscriptions authorized, we need continue to check the authorization for the topic resource at the line 145 for `checkAuthorization`.